core/src/main/java/net/sourceforge/jnlp/cache/cache/Cache.java

```diff
@@ -0,0 +1,173 @@
+     * @throws IllegalArgumentException if the resource is not cacheable
+     */
+    public static File getCacheFile(final URL resource, final VersionId version) {
+        if (!CacheUtil.isCacheable(resource)) {
+            throw new IllegalArgumentException(resource + " is not a cacheable resource");
+        }
+        return CacheLRUWrapper.getInstance().getOrCreateCacheFile(resource, version);
+    }
+
+    public static void addToCache(ResourceInfo infoFromRemote, InputStream unpackedStream) throws IOException {
+        if (!CacheUtil.isCacheable(infoFromRemote.getResourceHref())) {
+            throw new IllegalArgumentException(infoFromRemote.getResourceHref() + " is not a cacheable resource");
+        }
+        CacheLRUWrapper.getInstance().addToCache(infoFromRemote, unpackedStream);
+    }
+
+    /**
+     * Invalidate the entry and make it eligible for removal.
+     *
+     * @param resource the resource {@link URL}
+     * @param version  the versions
+     * @return the newly created cache file
+     * @throws IllegalArgumentException if the resource is not cacheable
+     */
+    public static File replaceExistingCacheFile(final URL resource, final VersionId version) {
+        if (!CacheUtil.isCacheable(resource)) {
+            throw new IllegalArgumentException(resource + " is not a cacheable resource");
+        }
+        return CacheLRUWrapper.getInstance().replaceExistingCacheFile(resource, version);
+    }
+
+    public static void deleteFromCache(ResourceInfo info) {
+        CacheLRUWrapper.getInstance().deleteFromCache(info.getResourceHref(), info.getVersion());
+    }
+
+    public static void deleteFromCache(URL resource, VersionString version) {
+        if (!CacheUtil.isCacheable(resource)) {
+            throw new IllegalArgumentException(resource + " is not a cacheable resource");
+        }
+        CacheLRUWrapper.getInstance().deleteFromCache(resource, version);
+    }
+
+    /**
+     * Returns whether there is a version of the URL contents in the
+     * cache.
+     *
+     * @param resource the resource {@link URL}
+     * @param version  the versions to check for
+     * @return whether the cache contains the version
+     * @throws IllegalArgumentException if the resource is not cacheable
+     */
+    public static boolean isCached(final URL resource, final VersionId version) {
+        if (!CacheUtil.isCacheable(resource)) {
+            throw new IllegalArgumentException(resource + " is not a cacheable resource");
+        }
+
+        return CacheLRUWrapper.getInstance().isCached(resource, version);
+    }
+
+    public static ResourceInfo getInfo(final URL resource, final VersionId version) {
+        if (!CacheUtil.isCacheable(resource)) {
+            throw new IllegalArgumentException(resource + " is not a cacheable resource");
+        }
+
+        return CacheLRUWrapper.getInstance().getResourceInfo(resource, version).orElse(null);
```

Comment:
CacheLRUWrapper <- Please rename the class since nobody knows what LRU stands for